sandbox/plc4c/examples/hello-subsciptions/src/hello_subscriptions.c

```diff
@@ -0,0 +1,264 @@
+
+    // Give plc4c a chance to do something.
+    // This is where all I/O is being done.
+    if (plc4c_system_loop(system) != OK) {
+      printf("ERROR in the system loop\n");
+      break;
+    }
+
+    // Depending on the current state, implement some logic.
+    switch (state) {
+      case CONNECTING: {
+        // Check if the connection is established:
+        if (plc4c_connection_get_connected(connection)) {
+          printf("SUCCESS\n");
+          state = CONNECTED;
+        } else if (plc4c_connection_has_error(connection)) {
+          printf("FAILED\n");
+          return -1;
+        }
+        break;
+      }
+      case CONNECTED: {
+        // Create a new read-request.
+        printf("Preparing a read-request ... ");
+
+        result = plc4c_connection_create_subscription_request(
+            connection, &subscription_request);
+        if (result != OK) {
+          printf("FAILED\n");
+          return -1;
+        }
+        printf("SUCCESS\n");
+
+        printf("Adding cyclic item for 'RANDOM/foo:INTEGER' ... ");
+        result = plc4c_subscription_request_add_cyclic_item(
+            subscription_request, "RANDOM/foo:INTEGER", 500);
+        if (result != OK) {
+          printf("FAILED\n");
+          return -1;
+        }
+        printf("SUCCESS\n");
+
+        // Execute the read-request.
+        printf("Executing a read-request ... ");
+        result = plc4c_subscription_request_execute(
+            subscription_request, &subscription_request_execution);
+        if (result != OK) {
+          printf("FAILED\n");
+          return -1;
+        } else {
+          state = SUBSCRIPTION_REQUEST_SENT;
+        }
+        break;
+      }
+        // Wait until the read-request execution is finished.
+      case SUBSCRIPTION_REQUEST_SENT: {
+        if (plc4c_subscription_request_execution_check_finished_successfully(
+                subscription_request_execution)) {
+          printf("SUCCESS\n");
+          state = SUBSCRIPTION_RESPONSE_RECEIVED;
+        } else if (
+            plc4c_subscription_request_execution_check_finished_with_error(
+                subscription_request_execution)) {
+          printf("FAILED\n");
+          return -1;
+        }
+        break;
+      }
+      case SUBSCRIPTION_RESPONSE_RECEIVED: {
+        // Get the response for the given subscription-request.
+        plc4c_subscription_response *subscription_response =
+            plc4c_subscription_request_execution_get_response(
+                subscription_request_execution);
+        if (subscription_response == NULL) {
+          printf("FAILED (No Response)\n");
+          return -1;
+        }
+
+        // Iterate over all returned items.
+        plc4c_list_element *cur_element =
+            plc4c_utils_list_head(subscription_response->response_items);
+        while (cur_element != NULL) {
+          plc4c_response_subscription_item *subscription_item_item =
+              cur_element->value;
+
```

Comment:
I think there would be value, in letting the caller set a 'cookie' or identifier in the subscription that they could check when they got it back.
It may not be a char* string that they have to strcmp.  This is common.

Reply:
No ... in this case we only have one subscription item, so it saves the subscription handle and then goes to the next step and waits till it receives 10 events and then exits ...